memory/service.go

```diff
@@ -61,19 +88,357 @@ func init() {
 			return nil, errors.Wrapf(err, "failed to open sqlite database at %s", conf.SqlitePath)
 		}
 
-		if err := db.AutoMigrate(&AgentContext{}); err != nil {
+		// Get the underlying sql.DB for sqlite-vec operations
+		rawDB, err := db.DB()
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to get raw database connection")
+		}
+
+		// Auto-migrate GORM entities
+		if err := db.AutoMigrate(&AgentContext{}, &entity.Knowledge{}); err != nil {
 			return nil, errors.Wrapf(err, "failed to auto-migrate sqlite database at %s", conf.SqlitePath)
 		}
+
 		c.RegisterOnShutdown(func(_ context.Context) {
-			db, err := db.DB()
-			if err != nil {
-				logger.Warn("failed to get database connection", slog.Any("error", err))
-			}
-			if err := db.Close(); err != nil {
+			if err := rawDB.Close(); err != nil {
 				logger.Warn("failed to close database connection", slog.Any("error", err))
 			}
 		})
 
-		return &SqliteService{db: db}, nil
+		// Create embedder for RAG functionality
+		genkit := din.MustGet[*genkit.Genkit](c, mygenkit.Key)
+		embedder := NewGenkitEmbedder(genkit)
+
+		service := &SqliteService{
+			db:           db,
+			rawDB:        rawDB,
+			embedder:     embedder,
+			vecExtLoaded: conf.VectorEnabled,
+		}
+
+		// Verify sqlite-vec is working and create vector table (only if vector functionality is enabled)
+		if conf.VectorEnabled {
+			if err := service.verifyAndCreateVectorTable(); err != nil {
+				return nil, errors.Wrapf(err, "failed to initialize sqlite-vec")
+			}
+		}
+
+		return service, nil
 	})
 }
+
+// initializeVectorFunctionality initializes sqlite-vec functionality on-demand
+func (s *SqliteService) initializeVectorFunctionality() error {
+	// Initialize sqlite-vec extension
+	sqlite_vec.Auto()
+
+	// Verify and create vector table
+	if err := s.verifyAndCreateVectorTable(); err != nil {
+		return errors.Wrapf(err, "failed to initialize vector table")
+	}
+
+	// Mark vector functionality as loaded
+	s.vecExtLoaded = true
+	return nil
+}
+
+// verifyAndCreateVectorTable verifies sqlite-vec is working and creates the vector table
+func (s *SqliteService) verifyAndCreateVectorTable() error {
+	// Verify sqlite-vec is loaded by checking vec_version()
+	var sqliteVersion, vecVersion string
+	err := s.rawDB.QueryRow("SELECT sqlite_version(), vec_version()").Scan(&sqliteVersion, &vecVersion)
```

Comment:
I think you can use `s.db.Exec()` instead of `s.rawDb`. Direct using sqlDB in gorm is not that the db connection managed by connection pool.

Reply:
@jcooky I've remove the rawDB references. Could you take a look again? thanks!